### PR TITLE
drivers: flash: QSPI flash driver for nRF52 SoC

### PR DESCRIPTION
**Pull request by findlayfeng**

### CODE REVIEW

**Comment by findlayfeng** [comment]
@nvlsianpu

**Comment by findlayfeng** [comment]
Format problem I adjusted on Monday

**Comment by nvlsianpu** [comment]
cc @kl-cruz

**Review by kl-cruz — COMMENTED** [review]
Overall conceptually looks good. Thanks for this!
I would consider to switch to IRQ mode and handle it internally in this driver (desing simple non-blocking mode) and add to DT configuration of the memory device to provide configuration and make code (switching to quad mode) more generic.

**Comment by findlayfeng** [comment]
@kl-cruz 
Thanks for the comment, I think the qspi device of nrf52840 is not an independent qspi, more accurately a flash controller.
I don't know what other Soc is like. The qspi of nrf52840 does not fully implement all the access methods in sfdp.
So my idea is that this is a flash device, you need to specify the flash chip.

File: drivers/flash/qspi_flash_nrf.c [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+#define LOG_DOMAIN "qspi_flash_nrfx_qspi"
+#define LOG_LEVEL CONFIG_SPI_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(qspi_flash_nrfx_qspi);
+
+struct qspi_flash_data {
+	bool write_protection_sw;
+};
+
+#define is_word_aligned(__addr) ((POINTER_TO_UINT(__addr) & 0x3) == 0)
```
[the remark this answers is not in the document]
Comment (nvlsianpu) [reply drivers/flash/qspi_flash_nrf.c]: >Maybe it could be good idea to move it to util?

good point. In self-commit pleas.

**Comment by thedjnK** [comment]
What is the status of this pull request?

**Comment by thedjnK** [comment]
@findlayfeng seems to be a timing issue with this code, with a loop that reads out large sections e.g. 8KB 32 bytes at a time, the first few runs will return correct data, then subsequent loops return more and more garbage, e.g. this data should be entirely ASCII: ribu�e�e�i�/�e�o�p�l�/�n�l�s�/�l�er th�s d�t�.��

Putting in a delay to the loop returns the correct data, however this is not desired operation.

**Comment by nvlsianpu** [comment]
looks like orphaned

**Pull request closed without merging**